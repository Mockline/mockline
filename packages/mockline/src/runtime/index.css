@import "tailwindcss";
@import "assets/css/animation.css";

@variant dark (&:where(.dark, .dark *));

@plugin "@tailwindcss/typography";

@theme {
  --color-neutral: var(--ui-bg);
  --color-neutral-elevated: var(--ui-bg-elevated);
  --color-border: var(--ui-border);
}

@layer base {
  :root {
    color-scheme: light;

    --ui-text-dimmed: var(--ui-color-neutral-400);
    --ui-text-muted: var(--ui-color-neutral-500);
    --ui-text-toned: var(--ui-color-neutral-600);
    --ui-text: var(--ui-color-neutral-700);
    --ui-text-highlighted: var(--ui-color-neutral-900);

    --ui-bg: var(--color-white);
    --ui-bg-elevated: var(--ui-color-neutral-100);
    --ui-bg-accented: var(--ui-color-neutral-200);
    --ui-bg-inverted: var(--ui-color-neutral-900);

    --ui-border: var(--ui-color-neutral-200);
    --ui-border-accented: var(--ui-color-neutral-300);
    --ui-border-inverted: var(--ui-color-neutral-900);

    --ui-radius: var(--radius);

    /*--ui-slate: var(--color-slate-500);
    --ui-gray: var(--color-gray-500);
    --ui-zinc: var(--color-zinc-500);
    --ui-stone: var(--color-stone-500);
    --ui-neutral: var(--color-neutral-500);
    --ui-red: var(--color-red-500);
    --ui-orange: var(--color-orange-500);
    --ui-amber: var(--color-amber-500);
    --ui-yellow: var(--color-yellow-500);
    --ui-lime: var(--color-lime-500);
    --ui-green: var(--color-green-500);
    --ui-emerald: var(--color-emerald-500);
    --ui-teal: var(--color-teal-500);
    --ui-cyan: var(--color-cyan-500);
    --ui-sky: var(--color-sky-500);
    --ui-blue: var(--color-blue-500);
    --ui-indigo: var(--color-indigo-500);
    --ui-violet: var(--color-violet-500);
    --ui-purple: var(--color-purple-500);
    --ui-fuchsia: var(--color-fuchsia-500);
    --ui-pink: var(--color-pink-500);
    --ui-rose: var(--color-rose-500);*/
  }

  .dark {
    color-scheme: dark;

    --ui-text-dimmed: var(--ui-color-neutral-500);
    --ui-text-muted: var(--ui-color-neutral-400);
    --ui-text-toned: var(--ui-color-neutral-300);
    --ui-text: var(--ui-color-neutral-200);
    --ui-text-highlighted: var(--color-white);

    --ui-bg: var(--ui-color-neutral-900);
    --ui-bg-elevated: var(--ui-color-neutral-800);
    --ui-bg-accented: var(--ui-color-neutral-700);
    --ui-bg-inverted: var(--color-white);

    --ui-border: var(--ui-color-neutral-800);
    --ui-border-accented: var(--ui-color-neutral-700);
    --ui-border-inverted: var(--color-white);
  }
}

