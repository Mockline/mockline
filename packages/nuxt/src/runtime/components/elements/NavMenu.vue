<script setup lang="ts">
import { tv, type VariantProps } from 'tailwind-variants'
import { computed } from 'vue'
import { twMerge } from 'tailwind-merge'
import {
  NavigationMenuContent,
  NavigationMenuIndicator,
  NavigationMenuItem,
  NavigationMenuLink,
  NavigationMenuList,
  NavigationMenuRoot,
  NavigationMenuTrigger,
  NavigationMenuViewport,
} from 'radix-vue'

const navMenuStyle = tv({
  slots: {
    content: 'absolute left-0 top-0 w-full data-[motion=from-end]:animate-[enter-from-right_200ms_ease] data-[motion=from-start]:animate-[enter-from-left_200ms_ease] data-[motion=to-end]:animate-[exit-to-right_200ms_ease] data-[motion=to-start]:animate-[exit-to-left_200ms_ease] sm:w-auto',
    viewport: 'bg-neutral border border-[var(--ui-border)] relative mt-[10px] h-[var(--radix-navigation-menu-viewport-height)] w-full origin-[top_center] overflow-hidden rounded-[10px] transition-[width,_height] duration-300 data-[state=closed]:animate-[scale-out_200ms_ease] data-[state=open]:animate-[scale-in_200ms_ease] sm:w-[var(--radix-navigation-menu-viewport-width)]'
  },
  variants: {
    color: {
      primary: 'text-primary-500 hover:bg-[var(--ui-bg-accented)] focus:shadow-[var(--ui-bg-inverted)]',
      white: 'hover:bg-[var(--ui-bg-accented)] focus:shadow-[var(--ui-bg-inverted)] text-neutral-700',
    },
  },
  defaultVariants: {
    color: 'primary',
  },
})

export type NavMenuProps = {
  orientation?: 'horizontal' | 'vertical'
  color?: VariantProps<typeof navMenuStyle>['color'];
  items: {
    title: string
    path: string
    slot?: string
  }[]
}

const props = withDefaults(defineProps<NavMenuProps>(), {
  orientation: 'horizontal',
})

const navTitleStyle = computed(() => {
  return twMerge([
    'group flex select-none items-center justify-between gap-[2px] rounded-[4px] px-3 py-2 text-[15px] font-medium leading-none outline-none focus:shadow-[0_0_0_2px]',
    navMenuStyle(),
  ])
})
</script>

<template>
  <NavigationMenuRoot class="relative z-[1] flex w-full justify-center" :orientation>
    <NavigationMenuList class="bg-neutral border-[var(--ui-border)] m-0 flex list-none items-center justify-between gap-8 rounded-lg border px-4 py-1.5 shadow-lg">
      <slot name="leading" />
      <div class="flex list-none items-center justify-between">
        <NavigationMenuItem v-for="item in items" :key="item.title">
          <NavigationMenuTrigger
            v-if="item.slot"
            :class="navTitleStyle"
          >
            {{ item.title }}
          </NavigationMenuTrigger>
          <NavigationMenuContent
            v-if="item.slot"
            :class="navMenuStyle.slots.content"
          >
            <slot :name="item.title.toLowerCase()" />
          </NavigationMenuContent>
          <NavigationMenuLink v-if="!item.slot" :href="item.path" :class="navTitleStyle">
            {{ item.title }}
          </NavigationMenuLink>
        </NavigationMenuItem>
      </div>

      <NavigationMenuIndicator />
      <slot name="trailing" />
    </NavigationMenuList>

    <div class="perspective-[2000px] absolute left-0 top-full flex w-full justify-center">
      <NavigationMenuViewport :class="navMenuStyle.slots.viewport" />
    </div>
  </NavigationMenuRoot>
</template>
