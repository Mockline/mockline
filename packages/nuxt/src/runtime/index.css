@import "@mockline/themes";
@import "./mockline.css";
