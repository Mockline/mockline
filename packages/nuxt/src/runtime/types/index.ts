// Elements
export * from '../components/Button.vue'
export * from '../components/NavMenu.vue'

// Layout
export * from '../components/App.vue'
export * from '../components/Aside.vue'
export * from '../components/Card.vue'
export * from '../components/Header.vue'
export * from '../components/Main.vue'
export * from '../components/Page.vue'
export * from '../components/PageBody.vue'
export * from '../components/PageHeader.vue'

// Settings
export * from '../components/ThemeToggle.vue'

// Overlays
export * from '../components/Toasts.vue'

// Content
export * from '../components/content/ContentNavigationTree.vue'
export * from '../components/content/ContentSurround.vue'
export * from '../components/content/ContentToc.vue'
export * from '../components/content/ContentTocLinks.vue'

// Global
export * from '../components/prose/Callout.vue'
export * from '../components/prose/CodeGroup.vue'
export * from '../components/prose/ProseCode.vue'
export * from '../components/prose/CodeButton.vue'
