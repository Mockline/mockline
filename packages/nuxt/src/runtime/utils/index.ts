export function transformSlot(slot: any, index: number): any {
  if (typeof slot.type === 'symbol') {
    return slot.children?.map(transformSlot)
  }
  return {
    label: slot.props?.filename || slot.props?.label || `${index}`,
    icon: slot.props?.icon,
    component: slot
  }
}

export * from './useComponent'
