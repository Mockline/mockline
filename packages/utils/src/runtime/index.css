@import "tailwindcss";

@variant dark (&:where(.dark, .dark *));

@plugin "@tailwindcss/typography";
