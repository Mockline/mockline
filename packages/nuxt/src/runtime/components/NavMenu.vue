<script setup lang="ts">
import { tv, type VariantProps } from 'tailwind-variants'
import { computed } from 'vue'
import { twMerge } from 'tailwind-merge'
import {
  NavigationMenuContent,
  NavigationMenuIndicator,
  NavigationMenuItem,
  NavigationMenuLink,
  NavigationMenuList,
  NavigationMenuRoot,
  NavigationMenuTrigger,
  NavigationMenuViewport,
} from 'reka-ui'
import type { NavMenuProps } from '@mockline/themes'
import { useComponent } from '#mockline/utils/useComponent'

const props = withDefaults(defineProps<NavMenuProps>(), {
  orientation: 'horizontal',
})

const { getClasses } = useComponent('navMenu', props)
</script>

<template>
  TODO
<!--  <NavigationMenuRoot class="relative z-[1] flex w-full justify-center" :orientation>
    <NavigationMenuList class="bg-neutral border-[var(&#45;&#45;ui-border)] m-0 flex list-none items-center justify-between gap-8 rounded-lg border px-4 py-1.5 shadow-lg">
      <slot name="leading" />
      <div class="flex list-none items-center justify-between">
        <NavigationMenuItem v-for="item in items" :key="item.label">
          <NavigationMenuTrigger
            v-if="item.slot"
            :class="navTitleStyle"
          >
            {{ item.title }}
          </NavigationMenuTrigger>
          <NavigationMenuContent
            v-if="item.slot"
            :class="navMenuStyle.slots.content"
          >
            <slot :name="item.title.toLowerCase()" />
          </NavigationMenuContent>
          <NavigationMenuLink v-if="!item.slot" :href="item.path" :class="navTitleStyle">
            {{ item.title }}
          </NavigationMenuLink>
        </NavigationMenuItem>
      </div>

      <NavigationMenuIndicator />
      <slot name="trailing" />
    </NavigationMenuList>

    <div class="perspective-[2000px] absolute left-0 top-full flex w-full justify-center">
      <NavigationMenuViewport :class="navMenuStyle.slots.viewport" />
    </div>
  </NavigationMenuRoot>-->
</template>
