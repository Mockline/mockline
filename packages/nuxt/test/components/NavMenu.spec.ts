import { describe, it, expect } from 'vitest'
import { renderSuspended } from '@nuxt/test-utils/runtime'
import NavMenu from '../../src/runtime/components/NavMenu.vue'

describe('NavMenu', () => {
  it('can mount component', async () => {
    const html = await renderSuspended(NavMenu)
    expect(html.html()).toMatchSnapshot()
  })
})
