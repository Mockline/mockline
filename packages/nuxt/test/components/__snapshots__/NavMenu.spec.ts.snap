// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavMenu > renders base case correctly 1`] = `
" TODO
<!--  <NavigationMenuRoot class="relative z-[1] flex w-full justify-center" :orientation>
    <NavigationMenuList class="bg-neutral border-[var(&#45;&#45;ui-border)] m-0 flex list-none items-center justify-between gap-8 rounded-lg border px-4 py-1.5 shadow-lg">
      <slot name="leading" />
      <div class="flex list-none items-center justify-between">
        <NavigationMenuItem v-for="item in items" :key="item.label">
          <NavigationMenuTrigger
            v-if="item.slot"
            :class="navTitleStyle"
          >
            {{ item.title }}
          </NavigationMenuTrigger>
          <NavigationMenuContent
            v-if="item.slot"
            :class="navMenuStyle.slots.content"
          >
            <slot :name="item.title.toLowerCase()" />
          </NavigationMenuContent>
          <NavigationMenuLink v-if="!item.slot" :href="item.path" :class="navTitleStyle">
            {{ item.title }}
          </NavigationMenuLink>
        </NavigationMenuItem>
      </div>

      <NavigationMenuIndicator />
      <slot name="trailing" />
    </NavigationMenuList>

    <div class="perspective-[2000px] absolute left-0 top-full flex w-full justify-center">
      <NavigationMenuViewport :class="navMenuStyle.slots.viewport" />
    </div>
  </NavigationMenuRoot>-->"
`;

exports[`NavMenu > renders horizontal correctly 1`] = `
" TODO
<!--  <NavigationMenuRoot class="relative z-[1] flex w-full justify-center" :orientation>
    <NavigationMenuList class="bg-neutral border-[var(&#45;&#45;ui-border)] m-0 flex list-none items-center justify-between gap-8 rounded-lg border px-4 py-1.5 shadow-lg">
      <slot name="leading" />
      <div class="flex list-none items-center justify-between">
        <NavigationMenuItem v-for="item in items" :key="item.label">
          <NavigationMenuTrigger
            v-if="item.slot"
            :class="navTitleStyle"
          >
            {{ item.title }}
          </NavigationMenuTrigger>
          <NavigationMenuContent
            v-if="item.slot"
            :class="navMenuStyle.slots.content"
          >
            <slot :name="item.title.toLowerCase()" />
          </NavigationMenuContent>
          <NavigationMenuLink v-if="!item.slot" :href="item.path" :class="navTitleStyle">
            {{ item.title }}
          </NavigationMenuLink>
        </NavigationMenuItem>
      </div>

      <NavigationMenuIndicator />
      <slot name="trailing" />
    </NavigationMenuList>

    <div class="perspective-[2000px] absolute left-0 top-full flex w-full justify-center">
      <NavigationMenuViewport :class="navMenuStyle.slots.viewport" />
    </div>
  </NavigationMenuRoot>-->"
`;

exports[`NavMenu > renders vertical correctly 1`] = `
" TODO
<!--  <NavigationMenuRoot class="relative z-[1] flex w-full justify-center" :orientation>
    <NavigationMenuList class="bg-neutral border-[var(&#45;&#45;ui-border)] m-0 flex list-none items-center justify-between gap-8 rounded-lg border px-4 py-1.5 shadow-lg">
      <slot name="leading" />
      <div class="flex list-none items-center justify-between">
        <NavigationMenuItem v-for="item in items" :key="item.label">
          <NavigationMenuTrigger
            v-if="item.slot"
            :class="navTitleStyle"
          >
            {{ item.title }}
          </NavigationMenuTrigger>
          <NavigationMenuContent
            v-if="item.slot"
            :class="navMenuStyle.slots.content"
          >
            <slot :name="item.title.toLowerCase()" />
          </NavigationMenuContent>
          <NavigationMenuLink v-if="!item.slot" :href="item.path" :class="navTitleStyle">
            {{ item.title }}
          </NavigationMenuLink>
        </NavigationMenuItem>
      </div>

      <NavigationMenuIndicator />
      <slot name="trailing" />
    </NavigationMenuList>

    <div class="perspective-[2000px] absolute left-0 top-full flex w-full justify-center">
      <NavigationMenuViewport :class="navMenuStyle.slots.viewport" />
    </div>
  </NavigationMenuRoot>-->"
`;
